python/ray/data/read_api.py

```diff
@@ -160,10 +164,28 @@ def read_datasource(datasource: Datasource[T],
 
     read_tasks = datasource.prepare_read(parallelism, **read_args)
     context = DatasetContext.get_current()
+    stats_actor = get_or_create_stats_actor()
+    stats_uuid = uuid.uuid4()
 
-    def remote_read(task: ReadTask) -> MaybeBlockPartition:
+    def remote_read(i: int, task: ReadTask) -> MaybeBlockPartition:
         DatasetContext._set_current(context)
-        return task()
+        start_time, start_cpu = time.perf_counter(), time.process_time()
+        exec_stats = BlockExecStats()
```

Comment:
maybe `with exec_stats.timer()` for both cpu and wall time?

Reply:
That's a TODO to add a builder here.